frameworks/moveos-stdlib/sources/module_store.move

```diff
@@ -108,18 +108,21 @@ module moveos_std::module_store {
 
     /// Publish modules to the module object's storage
     /// Return true if the modules are upgraded
-    public(friend) fun publish_modules_internal(module_object: &mut Object<ModuleStore>, package_id: address, modules: vector<MoveModule>) : bool {
+    public(friend) fun publish_modules_internal(
+        module_object: &mut Object<ModuleStore>, package_id: address, modules: vector<MoveModule>
+    ) : bool {
         let i = 0;
         let len = vector::length(&modules);
         let (module_names, module_names_with_init_fn, indices) = move_module::sort_and_verify_modules(&modules, package_id);
 
+        let is_upgrade = true;
         if (!exists_package(module_object, package_id)) {
-            let package = object::add_object_field_with_id(module_object, package_id, Package {});
-            object::transfer_extend(package, tx_context::sender());
+            // TODO: should we transfer the Package object to tx sender or package id address?
+            create_package(module_object, package_id, tx_context::sender());
+            is_upgrade = false;
         };
         let package = borrow_mut_package(module_object, package_id);
```

Comment:
Check if the `Package` is frozen and abort the transaction with an error code from the module_store.